core/modules/node/node.block.inc

```diff
@@ -0,0 +1,163 @@
+<?php
+
+/**
+ *  NodeBlock  extends Block
+ *  
+ *  This class allows us to display a particular node in a block.
+ */
+class NodeBlock extends Block {
+  /**
+   *  Sets title text on draggable block panel in Layout builder.
+   */
+  function getAdminTitle() {
+    return t('Content block');
+  }
+
+  /**
+   * Return a preview for this block.
+   */
+  function getAdminPreview() {
+    return t('Displays content of node ID: @nid', array('@nid' => $this->settings['nid']));
+  }
+
+  /**
+   *  Sets block subject on block view.
+   */
+  function getTitle() {
+    $title = NULL;
+    $node = node_load($this->settings['nid']);
+    if ($this->settings['title_display'] === LAYOUT_TITLE_CUSTOM && $this->settings['title']) {
+      $title = check_plain($this->settings['title']);
+    }
+    elseif ($this->settings['title_display'] === LAYOUT_TITLE_DEFAULT) {
+      $title = $node->title;
+    }
+
+    if ($title && $this->settings['link_node_title']) {
+      $title = l($title, 'node/' . $node->nid);
+    }
+
+    return $title;
+  }
+
+  /**
+   *  Sets block content on block view.
+   */
+  function getContent() {
+    $node = node_load($this->settings['nid']);
+
+    // Prevent display of no access to the node.
+    if (!node_access('view', $node)) {
+      return;
+    }
+
+    // Use a clone of the node so we can hide the title safely.
+    $clone = clone($node);
+
+    // Hide node title if not required.
+    if (empty($this->settings['leave_node_title'])) {
+      $clone->title = '';
+    }
+
+    $content = node_view($clone, $this->settings['view_mode']);
+
+    // Hide links if they've been suppressed.
+    if (empty($this->settings['links'])) {
+      $content['links']['#access'] = FALSE;
+    }
+
+    // Add 'submitted by' settings.
+    if (!empty($this->settings['display_submitted'])) {
+      $content['#display_submitted'] = TRUE;
+    }
+
+    // Add theme hook suggestion.
+    $content['#theme'] = 'node__content-block-' . $this->settings['nid'];
+
+    return $content;
+  }
+
+  /**
+   *  Builds the block's settings configuration form.
+   */
+  function form(&$form, &$form_state) {
+    parent::form($form, $form_state);
+    $form['title_display']['title_display']['#options'][LAYOUT_TITLE_DEFAULT] = t('Use node title');
+    $form['title_display']['title_display']['#weight'] = -50;
+    $settings = $this->settings;
+
+    $form['nid'] = array(
+      '#title' => t('Enter the title or NID of a node'),
+      '#type' => 'textfield',
+      '#maxlength' => 512,
+      '#default_value' => !empty($settings['nid']) ? $settings['nid'] : NULL,
+      '#autocomplete_path' => 'node/autocomplete',
+    );
+
+    $form['display_submitted'] = array(
+      '#type' => 'checkbox',
+      '#default_value' => !empty($settings['display_submitted']),
+      '#title' => t('Show author and date information'),
+      '#description' => t('Advanced: if checked, display the node author and date information.'),
```

Comment:
I'm not sure this is really "advanced". I would use the same description we use on the node type page...

Hm, actually that text is not great either:
```
Author username and publish date will be displayed.
```

I'm not sure these description texts are adding much.